### PR TITLE
Add Black Box terminal & Marble library

### PR DESCRIPTION
**Pull request by schuelermine**
###### Description of changes

Added the Black Box terminal as `blackbox-terminal`.
Added the Marble library, a dependency of Black Box, as `libmarble`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/blackbox-terminal/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    librsvg
+  ];
+
+  meta = {
+    description = "A beautiful and simple GTK 4 terminal";
+    homepage = "https://gitlab.gnome.org/raggesilver/blackbox";
+    maintainers = [ lib.maintainers.anselmschueler ];
+    license = lib.licenses.gpl3;
+    platforms = lib.platforms.linux;
+  }
```
Comment (wegank) [inline pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:84]:
```suggestion
  };
```

Comment (schuelermine) [reply pkgs/applications/terminal-emulators/blackbox-terminal/default.nix]: Done

File: pkgs/applications/terminal-emulators/blackbox-terminal/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    pcre2
+    libxml2
+    librsvg
+  ];
+
+  meta = {
+    description = "A beautiful and simple GTK 4 terminal";
+    homepage = "https://gitlab.gnome.org/raggesilver/blackbox";
+    maintainers = [ lib.maintainers.anselmschueler ];
+    license = lib.licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:82]: `gpl3` is unclear, use `gpl3Plus` or `gpl3Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

File: pkgs/applications/terminal-emulators/blackbox-terminal/default.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+, wrapGAppsHook4
+, python3
+, qt5
+}:
+
+stdenv.mkDerivation rec {
+  pname = "blackbox-terminal";
+  version = "0.12.0";
+
+  strictDeps = false;
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:32]: This is the default behaviour.

File: pkgs/applications/terminal-emulators/blackbox-terminal/default.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+, vala
+, vte
+, json-glib
+, pcre2
+, libxml2
+, librsvg
+, cmake
+, wrapGAppsHook4
+, python3
+, qt5
```
Comment (jtojnar) [inline pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:25]: Not used.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -1699,6 +1699,48 @@ with pkgs;
...
+        sha256 = "sha256-bwnGz9nTxcGjTbIXSqL23XoWW4PdtrvrIDNn+Vb6c5g=";
+      };
+      propagatedBuildInputs = attrs.propagatedBuildInputs ++ [
+        gtk4
+      ];
+      mesonFlags = attrs.mesonFlags ++ [
+        "-Dgtk4=true"
+      ];
+    });
+  };
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix:1742]: Not really fond of further bloating `all-packages.nix`. Maybe we should just wait for #182618.

Comment (schuelermine) [reply pkgs/top-level/all-packages.nix:1742]: OK, sure

File: pkgs/applications/terminal-emulators/blackbox-terminal/default.nix (hunk 3 above)
Comment (schuelermine) [reply pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:32]: Yes, but the build breaks if it is enabled, which might be a future default.

Comment (jtojnar) [reply pkgs/applications/terminal-emulators/blackbox-terminal/default.nix:32]: Then there should probably be a comment explaining the build failure.

**Comment by schuelermine** [comment]
Upon further thinking, I think it’s better to wait with packaging this until it’s more stable. Right now the Nix build inexplicably crashes sometimes due to garbage padding values, and the patch used by @mitchmindtree’s packaging seems to fix only some of the instances of this. In particular, this version crashes most of the time when the window is right-clicked. Additionally, @jtojnar has pointed out that it’s bad practice to override versions of packages in all-packages.nix, and that a PR is currently pending that seeks to introduce the needed version to nixpkgs. It could also be expected that upon stabilization of Black Box the ddeveloper might consider merging Marble’s WIP branch.

**Pull request closed without merging**